ironrdp/src/basic_output/bitmap.rs

```diff
@@ -97,7 +97,13 @@ impl<'a> PduBufferParsing<'a> for BitmapData<'a> {
         }
 
         let compressed_data_header = if !compression_flags.contains(Compression::NOT_COMPRESSED) {
-            Some(CompressedDataHeader::from_buffer_consume(buffer)?)
+            let consumed_buffer = CompressedDataHeader::from_buffer_consume(buffer);
+            match consumed_buffer {
+                Ok(buffer) => Some(buffer),
+                Err(error) => {
+                    return Err(error);
+                },
+            }
```

Comment:
I think this is strictly equivalent to using the `?` operator as before. Am I missing something?

Reply:
You're correct. This can just be allowed to propagate up.

Reply:
Reverted changes